File: 0101-symmetric-tree/0101-symmetric-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isSymmetric(TreeNode* root) {
        if(root==NULL)return true;
        return fun(root->left,root->right);
        
    }
    
    bool fun(TreeNode *left,TreeNode *right){
        if(left==NULL && right==NULL)return true;
        if(left==NULL || right==NULL)return false;
        if(left->val==right->val){
        return    fun(left->left,right->right) && fun(left->right,right->left);
        }
        return false;
    }
    
    
    
};

File: 0101-symmetric-tree/0101-symmetric-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
     if(root==null)return true;
       return fun(root.left,root.right); 
    }
    
    boolean fun(TreeNode left,TreeNode right){
        if(left==null && right==null)return true;
        if(left==null || right==null)return false;
        if(left.val==right.val){
            return fun(left.left,right.right) && fun(left.right,right.left);
            
        }
        return false;
        
    }
    
    
}